SYCL/GroupAlgorithm/exclusive_scan_sycl2020.cpp

```diff
@@ -0,0 +1,164 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -I . -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+// TODO: enable compile+runtime checks for operations defined in SPIR-V 1.3.
+// That requires either adding a switch to clang (-spirv-max-version=1.3) or
+// raising the spirv version from 1.1. to 1.3 for spirv translator
+// unconditionally. Using operators specific for spirv 1.3 and higher with
+// -spirv-max-version=1.1 being set by default causes assert/check fails
+// in spirv translator.
+// RUNx: %clangxx -fsycl -fsycl-targets=%sycl_triple -DSPIRV_1_3 %s -I . -o \
+   %t13.out
+
+#include "support.h"
+#include <CL/sycl.hpp>
+#include <algorithm>
+#include <cassert>
+#include <limits>
+#include <numeric>
+#include <vector>
+using namespace sycl;
+
+template <class SpecializationKernelName, int TestNumber>
+class exclusive_scan_kernel;
+
+// std::exclusive_scan isn't implemented yet, so use serial implementation
+// instead
```

Comment:
Now that we compile everything with C++17, can we replace this with `std::exclusive_scan`?

Reply:
I changed using of emu::exclusive scan to std::exclusive_scan

Reply:
It seems that Jenkins pre-ci linux/cuda compiles with c++14. Tests are failing with message "no member named 'exclusive_scan' in namespace 'std'"

Reply:
@maximdimakov, -fsycl triggers -std=c++17, so it should be compiled with c++17. Looks strange, does it reproduce locally?

Reply:
I can not reproduce this problem locally. The errors occurs when I explicitly specify -std=c++14. Without -std flag or with      -std=c++17 the test passed

Reply:
Looks like the environment problem of CI machine, could be outdated libstdc++ which don't contain `std::inclusive_scan` and `std::exclusive_scan`. @tfzhu, can you please take a look on this issue and try to resolve it?

Reply:
Also tagging @bader to be aware about this CI problem.

Reply:
Was the problem resolved? If not, could you please leave a TODO here about using `std::exclusive_scan` later?